src/main/scala/util/Annotations.scala

```diff
@@ -39,3 +42,109 @@ case class RetimeModuleAnnotation(target: ModuleName) extends SingleTargetAnnota
 trait ShouldBeRetimed { self: RawModule =>
   chisel3.experimental.annotate(new ChiselAnnotation { def toFirrtl = RetimeModuleAnnotation(self.toNamed) })
 }
+
+
+case class RegFieldDescMappingAnnotation(
+  target: ModuleName,
+  regMappingSer: RegistersSer) extends SingleTargetAnnotation[ModuleName] {
+  def duplicate(n: ModuleName): RegFieldDescMappingAnnotation = this.copy(target = n)
+}
+
+object GenRegDescsAnno {
+
+  def makeRegMappingSer(rawModule: RawModule,
+    moduleName: String,
+    baseAddress: BigInt,
+    width: Int,
+    byteOffset: Int,
+    bitOffset: Int,
+    regField: RegField): RegFieldSer = {
+
+    val anonRegFieldName = s"unnamedRegField${byteOffset.toHexString}_${bitOffset}"
+    val regFieldName = regField.desc.map {_.name}.getOrElse("")
+    val selectedRegFieldName = if (regFieldName.isEmpty /* selectedName.isEmpty */) anonRegFieldName else regFieldName
```

Comment:
This can just be:
`val selectedRegFieldName = regField.desc.map(_.name).getOrElse(anonRegFieldName)`